### PR TITLE
chore: components d to h type docs

### PR DESCRIPTION
**Pull request by SiTaggart**
- chore: initial types export
- chore: new api pages

Description and useful links to discussions / issues / tickets

### Change One

Details of change...

### Change Two

Details of change...

## Checklist

- [ ] Work in a "Draft" branch whilst you are getting feedback to reduce Chromatic costs. Once you are ready for approval, convert the PR to "Ready to review"
- [ ] Ensure all `required` checks have passed
- [ ] Icon changes must have a product design review
- [ ] Website content changes must have a product design review
- [ ] At least two engineers have reviewed any code changes
- [ ] At least one engineering lead has reviewed any core package changes or repository infrastructure
- [ ] Website visual regression testing is run by adding `🕵🏻‍♀️ Run website visual regression` label

### CODE REVIEW

File: .changeset/flat-worms-explode.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@twilio-paste/heading': minor
```
Comment (nkrantz) [inline .changeset/flat-worms-explode.md:2]: why did you switch to minors for this PR?

Comment (SiTaggart) [reply .changeset/flat-worms-explode.md:2]: Muscle memory

Comment (nkrantz) [reply .changeset/flat-worms-explode.md:2]: 👍 sticking to patches in mine just wanted to make sure

Comment (SiTaggart) [reply .changeset/flat-worms-explode.md:2]: I'll swap them to patches once I get back to this

File: packages/paste-website/src/pages/components/data-grid/api.mdx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        storybookUrl: '/?path=/story/components-data-grid--plain-data-grid',
+      },
+    },
+  };
+};
+
+## Installation
+
+```bash
+yarn add @twilio-paste/core
```
Comment (serifluous) [inline packages/paste-website/src/pages/components/data-grid/api.mdx:43]: suggestion: This probably shouldn't just be @twilio-paste/core?

File: packages/paste-website/src/pages/components/data-grid/api.mdx
Comment (serifluous) [inline packages/paste-website/src/pages/components/data-grid/api.mdx:1]: todo: Hide "inherited props" disclosure if nothing's in it, like DataGridRow

Comment (nkrantz) [reply packages/paste-website/src/pages/components/data-grid/api.mdx:1]: I suggested the same thing but as simon pointed out, all documented components that are used by consumers should have inherited props. aka this is a types fix that needs to be made on DataGridRow

**Review by nkrantz — APPROVED** [review]

**Review by nkrantz — APPROVED** [review]

**Pull request merged**